models/api-key.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+const dynamoDb = new AWS.DynamoDB.DocumentClient();
+const response = require ('../helpers/response.js');
+
+module.exports.activate = (requestBody, callback) => {
+  const data = JSON.parse(requestBody);
+  
+  if ((!data.email) || typeof data.email !== 'string') {
+    response.returnError(400, 'email is required', callback);
+    return;
+  }
+  
+  if ((!data.apiKey) || typeof data.apiKey !== 'string') {
+    response.returnError(400, 'apiKey is required', callback);
+    return;
+  }
+  
+  getApiKeyByValue(data.apiKey, (error, apiKeyRecord) => {
```

Comment:
Other than extracting out some of what I'm doing in this `module.exports.activate` function (so that what's left is easier to read), I don't know of a good way to increase the readability of this. This syntax seems to be the direction that JavaScript coding is apparently headed.

Reply:
How about ...
    ...

    getApiKeyByValue(
        data.apiKey, 
        (error, apiKeyRecord) => {
            if (error) { 
                ...
            }
            ...
        }
    );
};